src/ring_theory/polynomial/eisenstein.lean

```diff
@@ -185,3 +187,94 @@ end is_domain
 end is_eisenstein_at
 
 end polynomial
+
+section is_integral
+
+variables {K : Type v} {L : Type z} {p : R} [comm_ring R] [field K] [field L]
+variables [algebra K L] [algebra R L] [algebra R K] [is_scalar_tower R K L] [is_separable K L]
+variables [is_domain R] [normalized_gcd_monoid R] [is_fraction_ring R K] [is_integrally_closed R]
+
+local notation `𝓟` := submodule.span R {p}
+
+open is_integrally_closed power_basis nat polynomial is_scalar_tower
+
+/-- Let `K` be the field of fraction of an integrally closed domain `R` and let `L` be a separable
+extension of `K`, generated by an integral power basis `B` such that the minimal polynomial of
+`B.gen` is Eisenstein. Given `z : L` integral over `R`, if `Q : polynomial R` is such that
+`aeval B.gen Q = p • z`, then `p ∣ Q.coeff 0`. -/
+lemma dvd_coeff_zero_of_aeval_eq_prime_smul_of_minpoly_is_eiseinstein_at {B : power_basis K L}
+  (hp : prime p) (hei : (minpoly R B.gen).is_eisenstein_at 𝓟) (hBint : is_integral R B.gen) {z : L}
+  {Q : polynomial R} (hQ : aeval B.gen Q = p • z) (hzint : is_integral R z) :
+  p ∣ Q.coeff 0 :=
+begin
+  -- First define some abbreviations.
+  letI := B.finite_dimensional,
+  let P := minpoly R B.gen,
+  set n := B.dim,
```

Comment:
Optional suggestion: would it simplify things to do `cases n` up here to dispense with the `n = 0` case and thus all the `n - 1`s below should just become `n`s?

Reply:
`n` is always positive, so I changed its definition to avoid all the `n-1`. Thanks for the suggestion!